src/Collection.php

```diff
@@ -424,10 +424,13 @@ public function shuffle(int $seed = null): self
      * @param callable $callable
      * @return static
      */
-    public function implode($glue = "", callable $callable = null): string
+    public function implode($glue = '', callable $callable = null): string
     {
-        $data = $callable ? $this->map($callable)->getArrayCopy() : $this->getArrayCopy();
-        return implode($glue, $data);
+        $array = $this->getArrayCopy();
+        if ($callable && is_callable($callable)) {
```

Comment:
I'm not sure `is_callable` is needed here as the typehint should enforce that?

Reply:
True...

Reply:
Updated